### ISSUE TITLE
Passing list[str] to sorted in context of a print arg disallowed in pyright 1.1.286

### ISSUE THREAD

**Issue by aschmied** [microsoft/pyright#4391]
**Describe the bug**
pyright disallows passing `list[str]` to `sorted` when the list is sorted in the context of a `print` arg

**To Reproduce**
Create a file "repro.py" with content:

```python
l: list[str] = []
print("{s}".format(s="\n".join(sorted(l))))
```

then do:

```
$ pyright repro.py 
```

Output:

```
No configuration file found.
No pyproject.toml file found.
stubPath /home/aschmied/typings is not a valid directory.
Assuming Python platform Linux
Searching for source files
Found 1 source file
pyright 1.1.286
/home/aschmied/repro.py
  /home/aschmied/repro.py:2:39 - error: Argument of type "list[str]" cannot be assigned to parameter "__iterable" of type "Iterable[SupportsRichComparisonT@sorted]" in function "sorted"
    "list[str]" is incompatible with "Iterable[LiteralString]"
      TypeVar "_T_co@Iterable" is covariant
        "str" is incompatible with "LiteralString" (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations 
Completed in 0.577sec
```

**Expected behavior**
I expected pyright to complete successfully

**Additional context**

```
$ pyright --version
pyright 1.1.286
```

**Comment by aschmied** [comment]
I can work around it by moving the call to `sorted` out of the print arg:

```python
l: list[str] = []
s = "\n".join(sorted(l))
print("{s}".format(s=s))
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
@lmazuel, if you'd like me to verify that the problem is fixed, please provide a self-contained minimal sample. Your sample references symbols that are not defined.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.287, which I just published. It will also be included in a future release of pylance.

**Comment by lmazuel** [comment]
@erictraut sorry you were too fast for me here 😅 thanks for the release!
